JoiValidationStrategy.js

```diff
@@ -1,43 +1,41 @@
 var Joi = require('joi');
 
-var formatErrors = function(result) {
-  if (result.error !== null) {
-    return result.error.details.reduce(function(memo, detail) {
-      if (!Array.isArray(memo[detail.path])) {
-        memo[detail.path] = [];
-      }
-      memo[detail.path].push(detail.message);
-      return memo;
-    }, {});
-  } else {
-    return {};
-  }
-};
-
-var joiValidation = function(state, schema) {
-  return Joi.validate(state, schema, {
-    abortEarly: false,
-    allowUnknown: true,
-  });
-};
-
 var JoiValidationStrategy = {
+  validate: function(joiSchema, data, key) {
+    joiSchema = joiSchema || {};
+    data = data || {};
+    var joiOptions = {
+      abortEarly: false,
+      allowUnknown: true,
+    };
 
-  validate: function(options) {
-    if (options && options.schema) {
-      var errors = formatErrors(joiValidation(options.state || {}, options.schema));
-      if (options.field) {
-        var result = {};
-        result[options.field] = errors[options.field];
-        return result;
-      } else {
-        return errors;
-      }
+    var errors = this.format(Joi.validate(data, joiSchema, joiOptions));
+    if (key === undefined) {
+      Object.keys(joiSchema).forEach(function(key) {
+        errors[key] = errors[key] || [];
+      });
+      Object.keys(data).forEach(function(key) {
+        errors[key] = errors[key] || [];
+      });
+      return errors;
+    } else {
+      return errors[key] || [];
```

Comment:
This will simply return a list of error messages for this key, I feel it should maintain the same results structure when provided a key or not. The below snippet code will maintain the structure of [key: [messages...]}
```
var result = {};
result[options.field] = errors[options.field];
return result;
```